test/issues/0217.js

```diff
@@ -5,7 +5,7 @@ var assert = require('assert');
 var yaml   = require('../../');
 
 
-test('Javascript functions are not folded', function () {
+test.skip('Javascript functions are not folded', function () {
```

Comment:
[As @dervus expected](https://github.com/nodeca/js-yaml/pull/215#issuecomment-150790606), the actual problem was the folded style dumper was buggy. It tried to break more-indented lines, which resulted in newlines being added. Folding more-indented lines is not allowed in YAML.

Now that the folder is fixed it should work for all string data regardless of content.
This line tests for a workaround that doesn't fix the root problem (see #222).

Although I think it's better that the root problem is fixed, I can add the workaround back in if desired.